src/serialize.rs

```diff
@@ -0,0 +1,43 @@
+mod vec252;
+
+use crate::args::SerializeArgs;
+use anyhow::Result;
+use cairo_felt::Felt252;
+use cairo_proof_parser::parse;
+use itertools::chain;
+use std::fs;
+use std::path::Path;
+use vec252::VecFelt252;
+
+pub fn serialize_proof(args: &SerializeArgs) -> Result<()> {
+    let proof_file = args.proof.clone();
+    let (config, public_input, unsent_commitment, witness) = parse_proof_file(&proof_file)?;
+
+    let proof = chain!(
+        config.into_iter(),
+        public_input.into_iter(),
+        unsent_commitment.into_iter(),
+        witness.into_iter()
+    );
+
+    let calldata = chain!(proof, std::iter::once(Felt252::from(1)));
```

Comment:
What's the magic "1" constant?

Reply:
Nice catch, added the code for specifying this. (Cairo 1)